### PR TITLE
community endpoints

### PR DESCRIPTION
**Pull request by drewvolz**
This pull request adds Reddit integration to the St. Olaf College API, including endpoints for fetching subreddit posts and post comments, along with comprehensive parsing logic and tests. The main changes introduce the ability to fetch, parse, and serve Reddit content, with robust handling for both Atom (RSS) and JSON formats, and include tests to ensure correctness.

**Reddit API Integration and Endpoints:**

* Added new API endpoints `/reddit/posts/:subreddit` and `/reddit/comments` to fetch subreddit posts and comments, respectively, using new handler functions in `reddit.ts`.
* Registered the new Reddit endpoints in the main API router.

**Reddit Fetching and Parsing Logic:**

* Implemented `fetchRedditPosts` and `fetchRedditComments` in `feeds/reddit.ts`, including Atom and JSON parsing, comment tree construction, and error handling with Sentry.
* Added OAuth token management in `reddit-auth.ts` for authenticated Reddit API access, with in-memory caching and environment variable configuration.

**Testing and Utilities:**

* Added comprehensive tests for post and comment parsing, comment tree building, and edge cases in `feeds/reddit.test.ts`.
* Updated test setup to load environment variables from `.env` files.

### CODE REVIEW

**Comment by hawkrives** [comment]
Looks like formatting issues

File: source/ccc-lib/reddit-auth.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+	const credentials = Buffer.from(`${clientId}:${clientSecret}`).toString('base64')
+	const response = await http
+		.post('https://www.reddit.com/api/v1/access_token', {
+			headers: {Authorization: `Basic ${credentials}`},
+			body: new URLSearchParams({grant_type: 'client_credentials'}),
+		})
+		.json<TokenResponse>()
+
+	// Refresh 5 minutes before expiry
```
Comment (hawkrives) [inline source/ccc-lib/reddit-auth.ts:41]: this is not a proactive refresh, right? we're subtracting five minutes from the lifetime of our token, so that a request that comes in after that five minutes will require a refresh.

but, uh, I guess I'm wondering why we do that? Why not just keep the original lifetime, and just refresh after it expires? Or give ourselves like a five second grace period or something?

Comment (drewvolz) [reply source/ccc-lib/reddit-auth.ts:41]: I'll rethink this

Comment (drewvolz) [reply source/ccc-lib/reddit-auth.ts:41]: I also haven't been using the API tokens/auth yet. If we keep this locked down to dev-only, we may not need to think about auth at the moment

**Review by hawkrives — APPROVED** [review]
